backend/src/index.ts

```diff
@@ -1211,13 +1212,124 @@ app.post(
       });
       return res.status(200).json(newKeyword);
     } catch (e) {
-      return res.status(400).json({ message: 'invalid keyword input' });
+      return res.status(400).json({ message: "Invalid keyword input." });
     }
   }
 );
 
-// - app.post("/user/keyword") - Associates a user with a keyword
-// - app.delete("/user/keyword") - Disassociates a user with a keyword
+// associates a user with a keyword
+app.post(
+  "/user/keyword", 
+  async (req: TypedRequest<keywordIdBody>, res: Response) => {
+    //get userid from session
+    const sessionFromDB = await validateSession(
+      req.session ? req.session : null
+    );
+    if (!sessionFromDB) {
+      return res.status(401).json({ message: "Invalid session provided." });
+    }
+    const userID = sessionFromDB.userId;
+
+    //Make sure keyword exists
+    const keywordExists = await prisma.keyword.findFirst({
+      where: {
+        id: req.body.keywordId,
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    if (!keywordExists) {
+      return res.status(400).json({ message: "Invalid keyword." });

```

Comment:
i would probably 404 here. 400 is generally used for a bad request, eg invalid input types and such, but if a keyword just doesn't exist (which I believe you're checking here), i think 404 makes sense as it was Not Found